sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -110,6 +111,8 @@ abstract class DataType extends AbstractDataType {
 @InterfaceStability.Stable
 object DataType {
 
+  def fromDDL(ddl: String): DataType = CatalystSqlParser.parseDataType(ddl)
```

Comment:
I think it's reasonable for `DataType.fromDDL` to also support table style schema like `a int, b long`. How about we put the try catch here and other places just need to call `DataType.fromDDL`?